fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -355,6 +355,11 @@ ThreadFuture<Reference<ICluster>> DLApi::createCluster(const char *clusterFilePa
 	});
 }
 
+void DLApi::addNetworkThreadCompletionHook(void (*hook)(void*), void *hook_parameter) {
+	// All completion hooks are handled on the local client
+	throw unsupported_operation();
```

Comment:
I believe that this is the weirdness that caused us to put `addNetworkThreadCompletionHook` in `MultiVersionClientApi` but not `IClientApi` in the first place. I think there was also some concern that it wouldn't work with older clients. It looks like this doesn't have the second issue, so I guess that's fine. I'm not sure how I feel about the first, but maybe it's the least evil.

Reply:
Ideally, I think we'd just call straight through to the external library to handle completion hook, but as you say they may not support it. 

However, it might be the case that we don't run the network on clients that don't support the specified API version, and this function shouldn't be called prior to whatever API version it was introduced in. So maybe we can actually get away with implementing it that way.

Reply:
The above doesn't work because this function can be called even if using older API versions. I instead decided to change the implementation so that DLApi was responsible for managing callbacks at the external level.